app/controllers/admin/split_details_controller.rb

```diff
@@ -1,10 +1,10 @@
 class Admin::SplitDetailsController < AuthenticatedAdminController
   def edit
-    @split_detail = Split.find(params[:split_id]).detail
+    @split_detail = Split.includes(:variant_details).find(params[:split_id]).detail
   end
 
   def update
-    @split_detail = Split.find(params[:split_id]).detail
+    @split_detail = Split.includes(:variant_details).find(params[:split_id]).detail
```

Comment:
my bad, I read the view wrong and thought we were iterating over the splits and causing an n+1, the way this worked was totally fine since it's a single query per table.